rekor-v2-spec.md

```diff
@@ -0,0 +1,241 @@
+validate the signature. The service must store enough information in the Merkle
+tree leaf for an interested party to perform the same record validation to
+ensure entry auditability.
+
+### 4.2 HTTP and gRPC API
+
+The Rekor v2 write API is accessible over HTTP or gRPC, and the read-only tiles
+API and checkpoint are accessible over HTTP. Refer to the [protobuf
+specification](https://github.com/sigstore/rekor-tiles/tree/main/api/proto) for
+the exact schema, and the [clients
+guide](https://github.com/sigstore/rekor-tiles/blob/main/CLIENTS.md) for
+examples.
+
+### 4.3 Checkpoint Format
+
+The Merkle tree checkpoint, accessible via a GET request to the
+`/api/v2/checkpoint` endpoint, follows the [C2SP Transparency Log
+Checkpoints](https://github.com/C2SP/C2SP/blob/main/tlog-checkpoint.md)
+specification. The checkpoint MAY additionally include witness signatures as
+specified in the [Transparency Log Witness
+Protocol](https://github.com/C2SP/C2SP/blob/main/tlog-witness.md).
+
+## 5. Client Workflows
+
+### 5.1 Signers
+
+A client signing an artifact uploads the entry record, formatted as either a
+[HashedRekordRequestV002](https://github.com/sigstore/rekor-tiles/blob/f3cba09c2f92f1d2d5a7ca2b5694b66941a82d41/api/proto/rekor/v2/hashedrekord.proto#L31C9-L31C32)
+or
+[DSSERequestV002](https://github.com/sigstore/rekor-tiles/blob/f3cba09c2f92f1d2d5a7ca2b5694b66941a82d41/api/proto/rekor/v2/dsse.proto#L32C9-L32C24) to the
+`/api/v2/log/entries` (HTTP) or `dev.sigstore.rekor.v2.Rekor.CreateEntry` (gRPC)
+service. The client receives, as a response, a
+[TransparencyLogEntry](https://github.com/sigstore/protobuf-specs/blob/4df5baadcdb582a70c2bc032e042c0a218eb3841/protos/sigstore_rekor.proto#L94),
+which contains the latest checkpoint and an inclusion proof for the entry.
+Because entries are batched before being integrated, and because checkpoints are
+published only at regular intervals, the client may have to wait for a few
+seconds before receiving a response, depending on the configuration of the
+service.
+
+### 5.2 Artifact Verifiers
+
+A client verifying an artifact MUST NOT perform an online lookup of the entry.
+The signing client must provide the entry's checkpoint, inclusion proof,
+verification key and digest in the verification material it provides for the
+verifying client. The verifying client uses the material it is provided with and
+the public key of the log to verify the entry's inclusion in the log.
+
+### 5.3 Monitors
+
+A monitor is a client that monitors the contents of a transparency log to
+protect a given user against entry key compromise. If a key or identity is found
+to have been used in a signing event, it is done so in the open, and a monitor
+detects it and reports it to the user to whom the key or identity belongs. The
+monitor uses the tile entries API at `/api/v2/tile/entries` to tail the log for
+its contents. A monitor could also periodically verify the consistency of the
```

Comment:
Let's drop this last sentence, this blurs the line between a monitor and a witness.

Reply:
:+1: